lib/internal/abort_controller.js

```diff
@@ -372,18 +374,48 @@ ObjectDefineProperty(AbortSignal.prototype, SymbolToStringTag, {
 
 defineEventHandler(AbortSignal.prototype, 'abort');
 
+// https://dom.spec.whatwg.org/#dom-abortsignal-abort
 function abortSignal(signal, reason) {
+  // 1. If signal is aborted, then return.
   if (signal[kAborted]) return;
+
+  // 2. Set signal's abort reason to reason if it is given;
+  //    otherwise to a new "AbortError" DOMException.
   signal[kAborted] = true;
   signal[kReason] = reason;
+  // 3. Let dependentSignalsToAbort be a new list.
+  const dependentSignalsToAbort = [];
+  // 4. For each dependentSignal of signal's dependent signals:
+  if (signal[kDependantSignals]) {
+    // Equivalent to Set.prototype.forEach
+    SetPrototypeForEach(signal[kDependantSignals], (s) => {
```

Comment:
We're using a `SafeSet`, there's no point in using `SetPrototypeForEach`
```suggestion
  if (signal[kDependantSignals]) {
    signal[kDependantSignals].forEach((s) => {
```
What about using optional chaining btw?
```suggestion
  signal[kDependantSignals]?.forEach((s) => {
```

Reply:
makes sense. If primordials is unnecessary then optional chaining is better.